src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -868,6 +868,42 @@ void Lowering::LowerHWIntrinsicCmpOp(GenTreeHWIntrinsic* node, genTreeOps cmpOp)
     GenTree* op1 = node->Op(1);
     GenTree* op2 = node->Op(2);
 
+    // Optimize comparison against Vector64/128<>.Zero via UMAX:
+    //
+    //   bool eq = v == Vector128<integer>.Zero
+    //
+    // to:
+    //
+    //   bool eq = AdvSimd.Arm64.MaxAcross(v.AsByte()).ToScalar() == 0;
+    //
+    // Assume morph made get_Zero rhs if it was lhs
```

Comment:
Does morph actually swap the operands for `op_Equality` ? We should add a test doing the swap:
```
bool eq = Vector128<integer>.Zero == v
```

I am curious if the swap happens in morph. We recently are swapping the operands for `AdvSimd.CompareEqual` in lowering which makes me wonder if we should do it earlier.

Reply:
> I am curious if the swap happens in morph.

if it does not it probably makes sense to do - we do it for scalar expressions there